### PR TITLE
Add Capistrano 3 support

### LINKED ISSUE #17: Capistrano 3 support
Any plan to support capistrano 3 in the near future?

### PR DESCRIPTION
**Pull request by aleksandrs-ledovskis**
This commit introduces Capistrano 3+ support in a backwards compatible way (i.e. Capistrano 2 still works).

Also fixes bug [I think] @yhourdel spotted in #18

    undefined method `[]' for #<IO:<STDOUT>>

 by specifically referring to stdlib `Logger` instead of Capistrano's [one](https://github.com/capistrano/capistrano/blob/2.15.5/lib/capistrano/logger.rb#L79)

Fixes #18
Closes #17

### CODE REVIEW

**Comment by yhourdel** [comment]
Nice !

**Comment by j15e** [comment]
Wow much :rainbow: ! Thanks

File: lib/capistrano/s3.rb [hunk 1]
```diff
@@ -1,43 +1,10 @@
...
-      task :restart do; end
-    end
-  end
+require "capistrano"
+require "capistrano/s3/publisher"
+require "capistrano/s3/version"
+require "capistrano/s3/defaults"
+
+if Gem::Specification.find_by_name('capistrano').version >= Gem::Version.new('3.0.0')
+  load File.expand_path('../tasks/capistrano_3.rb', __FILE__)
```
Comment (j15e) [inline lib/capistrano/s3.rb]: Why not `require "capistrano/s3/capistrano_3"` here and below? Like I am not sure why you use 2 different loading method (`require_relative` vs `load + File.expand_path`)

File: README.md [hunk 2]
```diff
@@ -43,6 +43,11 @@ set :access_key_id, "CHANGETHIS"
 set :secret_access_key, "CHANGETHIS"
 ```
 
+### Capistrano 3
```
Comment (j15e) [inline README.md]: To further avoid confusion I would split the configuration for Capistrano 2 too (or change the default suggested configuration to Capistrano 3)

File: lib/capistrano/s3.rb (hunk 1 above)
Comment (aleksandrs-ledovskis) [inline lib/capistrano/s3.rb]: Using `require`/`require_relative` necessitate `Module` of some sort. Calling it on Capistrano's DSL is no go. Therefore the `load`.

Also this construct seems [pretty common](https://github.com/search?utf8=%E2%9C%93&q=Gem%3A%3ASpecification.find_by_name%28%27capistrano%27%29.version&type=Code&ref=searchresults)

Comment (j15e) [inline lib/capistrano/s3.rb]: Ok great!

File: README.md (hunk 2 above)
Comment (aleksandrs-ledovskis) [inline README.md]: Have updated README, take a look

**Comment by aleksandrs-ledovskis** [comment]
:fast_forward: bump?

**Comment by j15e** [comment]
I'll merge, sorry I completely forgot about that pull request. I am not frequently using this gem anymore. I'll add you as collaborator if you want to keep it up to date @aleksandrs-ledovskis.

**Comment by j15e** [comment]
Do you have a rubygems.org user so I could give you publish permission?

**Comment by aleksandrs-ledovskis** [comment]
@j15e 

Thanks for granting collaborator status. I'll merge this one shortly and look through the other PR's a bit later.

On RubyGems I'm [aleksandrs-ledovskis](https://rubygems.org/profiles/aleksandrs-ledovskis)

**Comment by j15e** [comment]
Ok done! Beware the gem was published with a signed key, you can follow the procedure here to keep it signed with a new key :

http://guides.rubygems.org/security/#building-gems

And add your public key to the repo : 

https://github.com/hooktstudios/capistrano-s3/tree/master/certs

**Pull request merged**